tests/test_backend.py

```diff
@@ -163,6 +215,75 @@ def test_cache_value_error(self):
         with self.assertRaises(ValueError):
             b.cache = 8
 
+    def test_fetch_client_not_provided(self):
+        """Test whether an NotImplementedError exception is thrown"""
+
+        b = Backend('test')
+
+        with self.assertRaises(NotImplementedError):
+            _ = [item for item in b.fetch(category="acme")]
+
+    def test_fetch_items(self):
+        """Test whether an NotImplementedError exception is thrown"""
+
+        b = Backend('test')
+
+        with self.assertRaises(NotImplementedError):
+            b.fetch_items()
+
+    def test_init_client(self):
+        """Test whether an NotImplementedError exception is thrown"""
+
+        b = Backend('test')
+
+        with self.assertRaises(NotImplementedError):
+            b.fetch_items()
+
+
+class TestBackendArchive(unittest.TestCase):
```

Comment:
I think this test should go in a different file that won't run. For instance, `base.py` in `tests` directory. In the same way, the test can be renamed to `TestCaseBackendArchive`.

The method `test_fetch_from_archive` won't receive any parameters but `self` (as other test methods). `backend` object will be called using `self.backend`. Derivated classes should define `setUp` method as follows (i.e Bugzilla):

```
class TestBugzillaBackendArchive(TestCaseBackendArchive):

    def setUp(self):
        super().setUp()
        self.backend = Bugzilla(archive=self,.archive, ...)

        # Setup here HTTP server using httpretty.
```

Reply:
You should define a `TestMockedBackendArchive` in this file too.